### ISSUE TITLE
[functionality] how about moment-timezone ?

### ISSUE THREAD

**Issue by jpvincent** [iamakulov/moment-locales-webpack-plugin#17]
Hi
Thank you for this brilliant plugin, it replaces the old hack we had until now :)

moment-timezone shares with moment a similar problem : the data comes packed with it, it's massive (901 Kb before gzip), and you can't choose just what you need.

There is Luxon, that is using the browser database, but it cant work on some browsers, including IE 11, and anyway it's a massive API change.

Here is what you have in the data : 
```console
more node_modules/moment-timezone/data/packed/latest.json
```

```json
{
        "version": "2018g",
        "zones": [
                "Africa/Abidjan|LMT GMT|g.8 0|01|-2ldXH.Q|48e5",
                "Africa/Accra|LMT GMT +0020|.Q 0 -k|012121212121212121212121212121212121212121212121|-26BbX.8 6tzX.8 MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE 1BAk MnE 1C0k MnE 1BAk MnE 1BAk MnE|41e5",

```

In my case, instead of all the possible timezones in the world, I just need one, but I would be crazy to code the timezone specific rules, especially given that they can change.
Another use cas would be to have only the America/* timezones for example.

Do you think it's possible to write a plugin that would strip the data, based on an expression ?

**Comment by gilmoreorless** [comment]
I wrote a custom Webpack helper a little while ago to do exactly this. A project only needed time zone definitions for news articles in 2 countries from the launch date (in 2017) onwards. Stripping it down to only what we needed took the moment-timezone data to 0.4KB gzipped.

I'll try to set up a new standalone Webpack plugin based on that helper script. (If I publicly commit to doing it via this comment, there's a better chance of me following through. 😉)

**Comment by iamakulov** [comment]
Hey! Optimizing moment-timezone would certainly be useful.

I think adding moment-timezone into _this_ plugin would complicate the user-facing API and the implementation – so it’s better to create a separate one. (“Do one thing and do it well”, you know.) I won’t be able to focus on this in the nearest time though, unfortunately – but if someone creates such plugin, I’d be happy to mention it in the readme!

**Comment by iamakulov** [comment]
@gilmoreorless Would you mind sharing your helper script? Before there’s a plugin for `moment-timezone`, the script might help folks who encounter the issue :)

**Comment by gilmoreorless** [comment]
@iamakulov Great idea. I've popped it up at https://gist.github.com/gilmoreorless/30d77f9fbb58623a1e4d5609b4b61d56

It uses a fragile and hard-coded method of intercepting the data, but it can be used as a starting point at least. 😃

**Comment by gilmoreorless** [comment]
Hi everyone. Amazingly for me and my habit of procrastination, I've followed through and turned that helper script into a proper webpack plugin: https://github.com/gilmoreorless/moment-timezone-data-webpack-plugin

I've switched over the commercial project I was working on to use the plugin instead of the custom script, and it seems to work fine. It's only been tested on that one project though, so let me know if you find any problems with it.
